README.md

```diff
@@ -1,71 +1,157 @@
+
+1. The value of the `mode` field must be valid (currently only "time" is accepted).
+2. The value of `defaultThreshold` must be a positive integer, and `defaultThreshold` cannot be omitted.
+3. There must be at least one rollout stage.
+4. The values of `percent` fields must be in non-decreasing order.
+5. The values of `percent` fields must be in the range [1, 99]. You do not need to specify 0 as the first stage, and you MUST not specify 100 as the final stage.
+6. Any optional `threshold` values, if specified, must be positive integers.
+
+## Example Workflow
+
+### Installing KCD
+
+Before installing and using KCD, you need to make sure that you have `go`, `kubectl`, and `ko` installed on your machine. Your cluster must also have Knative.
+
+To install KCD, run the following command in your terminal:
 
 ```sh
-export GOPATH="$HOME/go"
-export PATH="${PATH}:${GOPATH}/bin"
+ko apply -f config/
 ```
 
-### Clone to your machine
+This will create all the prerequisite custom resource definitions and deploy the KCD controller and webhook to your cluster. To verify that the controller and webhook have been correctly deployed, you can use the following command: 
+
+```sh
+kubectl get pod -n knative-serving
+```
 
-1. [Fork this repo](https://help.github.com/articles/fork-a-repo/) to your account
-2. Clone to your machine
+In the printout, if you can find two lines similar to the ones below, then your KCD controller and webhook are correctly deployed:
+
+```
+NAME                                                            READY   STATUS    RESTARTS   AGE
+continuous-delivery-webhook-677cf69bcd-k4lls                    1/1     Running   0          89s
+knative-continuous-delivery-dg8g6-deployment-6d97dd9869-r6hp9   1/2     Running   0          88s
+```
+
+### Applying your Policy or use a built-in Policy
+
+A few built-in policies have been provided in `example-policies/`. You can apply them by doing
 
 ```sh
-mkdir -p ${GOPATH}/src/github.com/googleinterns
-cd ${GOPATH}/src/github.com/googleinterns
-git clone git@github.com:${YOUR_GITHUB_USERNAME}/knative-continuous-delivery.git
-cd knative-continuous-delivery
-git remote add upstream https://github.com/googleinterns/knative-continuous-delivery.git
-git remote set-url --push upstream no_push
+ko apply -f example-policies
 ```
 
-`Upstream` will allow for you to [sync your fork](https://help.github.com/articles/syncing-a-fork/)
+Note that to actually use a built-in policy in a rollout, the relevant Service or Configuration must have the appropriate annotations to specify the name of the policy that it is using.
 
-### Building
+### Deploying your service
 
-`go build ./...`
+Suppose you have a Knative Service `default/my-app` in a file called `my-app.yaml`, then you can simply run the following command to deploy it:
 
-### Running unit tests
+```sh
+kubectl apply -f my-app.yaml
```

Comment:
what is my-app.yaml? This should tell a story of 

1. Install KCD
2. Add an example policy
3. Add a ksvc that clearly indicates the policy.

Reply:
This is the third subsection under "Example Workflow" (line 77). I described how to install KCD and apply policy on lines 79-111

Reply:
where does my-app come from? I think it is helpful to have an explicit service yaml that uses the basic.yaml policy.

Reply:
Oh I see what you mean. I fix it and resubmit